### ISSUE TITLE
-Wmissing-prototypes does not work

### ISSUE THREAD

**Issue by chengniansun** [llvm/llvm-project#18937]
|  |  |
| --- | --- |
| Bugzilla Link | [18563](https://llvm.org/bz18563) |
| Version | trunk |
| OS | Linux |

## Extended Description 
The function "bar()" does not have a prototype. But clang does not warn it. Please refer to the following steps.

$: cat no-prototype.c
```c++
static void bar();

static void foo(void) { bar(); }

void bar(void) {}

int main() {
  foo();
  return 0;
}
```
$: clang-trunk -Weverything no-prototype.c
$: clang-trunk -Wmissing-prototypes no-prototype.c
$: clang-trunk --version
clang version 3.5 (trunk 199549)
Target: x86_64-unknown-linux-gnu
Thread model: posix
$: gcc-trunk-warn no-prototype.c
no-prototype.c:5:6: warning: no previous prototype for ‘bar’ [-Wmissing-prototypes]
 void bar(void) {}
      ^
$:

**Comment by philnik777** [comment]
`-Wstrict-prototypes` warns that `static void bar()` isn't a prototype, and `bar` is `static`, so I don't think this is a problem.
